### PR TITLE
HMOS Library

### PR DESCRIPTION
**Pull request by SaiPolu**

### CODE REVIEW

File: entry/src/main/java/com/example/myapplication/slice/MainAbilitySlice.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        import ohos.agp.components.Image;
+        import ohos.agp.components.Slider;
+        import ohos.agp.components.Text;
+        import ohos.global.resource.Resource;
+
+public class MainAbilitySlice extends AbilitySlice {
+    @Override
+    public void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_ability_main);
```
Comment (sharathks84) [inline entry/src/main/java/com/example/myapplication/slice/MainAbilitySlice.java:10]: user doesnt have any control to handle this photoViewer.

File: photo_viewer/src/main/java/com/example/photo_viewer/Util.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    private static final int HILOG_TYPE = 3;
+
+    private static final int HILOG_DOMAIN = 0xD000F00;
+
+    private static final HiLogLabel LABEL = new HiLogLabel(HILOG_TYPE, HILOG_DOMAIN, "[Utils]");
+
+    private Util() {
+    }
+
+    public static Optional<PixelMap> preparePixelmap(final Resource resource) {
```
Comment (sharathks84) [inline photo_viewer/src/main/java/com/example/photo_viewer/Util.java]: There is no functionality for this method

File: photo_viewer/src/main/java/com/example/photo_viewer/Util.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        decodingOpts.desiredRegion = new Rect(0, 0, 0, 0);
+        decodingOpts.desiredPixelFormat = PixelFormat.ARGB_8888;
+        PixelMap decodePixelMap = null;
+        if (imageSource != null) {
+            decodePixelMap = imageSource.createPixelmap(decodingOpts);
+        }
+        return Optional.ofNullable(decodePixelMap);
+    }
+
+    private static byte[] readBytes(final Resource resource) {
```
Comment (sharathks84) [inline photo_viewer/src/main/java/com/example/photo_viewer/Util.java]: No functionality of this method

File: photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+        public Photoviewer(Context context, AttrSet attrSet) {
+                super(context, attrSet);
+                initAttr(attrSet);
+                setClickedListener(this);
+        }
+
+        private void initAttr(AttrSet attrSet) {
+
+                checkImagetype();
```
Comment (sharathks84) [inline photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java]: CheckImageType and its usages are not needed. Need to remove

File: photo_viewer/build.gradle [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apply plugin: 'com.huawei.ohos.library'
+//For instructions on signature configuration, see https://developer.harmonyos.com/en/docs/documentation/doc-guides/ide_debug_device-0000001053822404#EN-US_TOPIC_0000001154985555__section1112183053510
+ohos {
+    compileSdkVersion 5
+    defaultConfig {
+        compatibleSdkVersion 4
```
Comment (sharathks84) [inline photo_viewer/build.gradle:6]: compatibility should be 5 or 6 and compile is 6

File: photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import ohos.utils.net.Uri;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Timer;
+import java.util.TimerTask;
+
+
+public class Photoviewer extends Image implements Component.ClickedListener, ohos.agp.render.render3d.Component {
+
```
Comment (sharathks84) [inline photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java]: Resolve all the compiler warnings

File: photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        time.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                        AnimatorProperty animatorProperty = component.createAnimatorProperty();
+                        animatorProperty.scaleX(1).scaleY(1);
+                        animatorProperty.start();
+                }
+        }, 500);
+        time.schedule(new TimerTask() {
```
Comment (sharathks84) [inline photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java:28]: This animation works only works once, post that this 360 turn doesnt work

Comment (SaiPolu) [reply photo_viewer/src/main/java/com/example/photo_viewer/Photoviewer.java:28]: Resolved Sir

**Review by sharathks84 — APPROVED** [review]
OK, Approved

**Review by Gleelakrishna — APPROVED** [review]
ok

**Pull request merged**